### PR TITLE
Closes #6428 Bail-out of warmup if licence expired since more than 15 days

### LINKED ISSUE #6428: 3.16 - Bail-out of pre-warmup for LCP when license is expired for more than 15 days
### Context
For the 3.16 version, we need to prevent sending requests to the SaaS and stop preloading if the license is expired for more than 15 days. 
#### in `Engine\License\API\User`
- Add a new method to check if user is expired since more than 15 days
``` php
// Check if the license has been expired for more than 15 days
if ( $user->is_license_expired() && (time() - $license_expiration > 15 * 24 * 60 * 60) ) {
	return  false;
}
```
- Add test

#### in `Engine\Media\AboveTheFold\Warmup\Controller`
- Add a new dependency on the User class
- In the method that manages the ` fetch_links()`, add a check with the new user expired method above
- Update tests 

### Expected Behavior
The pre-warmup should not start if the license is expired for more than 15 days.

### Effort estimation:
 Effort [S]

### PR DESCRIPTION
**Pull request by Khadreal**
# Description

Fixes #6428

### User documentation

This won't allow user that their licence has expired past 15 days to use the pre-warmup feature

### Technical documentation

*Explain how this code works. Diagram & drawings are welcomed.*

## Type of change
- [x] New feature (non-breaking change which adds functionality).

## New dependencies
N/A

## Risks
*List possible performance & security issues or risks, and explain how they have been mitigated.*

# Checklists

## Feature validation

- [x] I validated all the Acceptance Criteria. If possible, provide sreenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Documentation

- [ ] I prepared the user documentation for the feature/enhancement and shared it in the PR or the GitHub issue.
- [ ] The user documentation covers new/changed entry points (endpoints, WP hooks, configuration files, ...).
- [ ] I prepared the technical documentation if needed, and shared it in the PR or the GitHub issue.

## Code style
- [x] I wrote self-explanatory code about what it does.
- [x] I wrote comments to explain why it does it.
- [x] I named variables and functions explicitely.
- [ ] I protected entry points against unexpected inputs.
- [x] I did not introduce unecessary complexity.
- [ ] I listed the introduced external dependencies explicitely on the PR.
- [ ] I validated the repo-specific guidelines from CONTRIBUTING.md.

## Observability
- [ ] I handled errors when needed.
- [ ] I wrote user-facing messages that are understandable and provide actionable feedbacks.
- [ ] I prepared ways to observe the implemented system (logs, data, etc.).

## Risks
- [ ] I explicitely mentioned performance risks in the PR.
- [ ] I explicitely mentioned security risks in the PR.

### CODE REVIEW

File: tests/Unit/inc/Engine/Media/AboveTheFold/WarmUp/Controller/fetchLinks.php [hunk 1]
```diff
@@ -6,26 +6,29 @@
...
+use WP_Rocket\Engine\License\API\User;
 use WP_Rocket\Engine\Media\AboveTheFold\WarmUp\Controller;
 use WP_Rocket\Tests\Unit\TestCase;
 
 /**
  * @covers \WP_Rocket\Engine\Media\AboveTheFold\WarmUp\Controller::fetch_links
  *
  * @group Media
- * @group AboveTheFold
+ * @group ATF
```
Comment (remyperona) [inline tests/Unit/inc/Engine/Media/AboveTheFold/WarmUp/Controller/fetchLinks.php]:
```suggestion
 * @group AboveTheFold
```

File: tests/Unit/inc/Engine/Media/AboveTheFold/WarmUp/Controller/fetchLinks.php [hunk 2]
```diff
@@ -36,39 +39,41 @@ protected function tearDown(): void {
...
+                ->andReturn( $config['response']['body'] );
+        }
 
-		if ( isset( $config['found_link'] ) && $config['found_link'] ) {
+        if ( isset( $config['found_link'] ) && $config['found_link'] ) {
 
-			$this->stubWpParseUrl();
+            Functions\when( 'wp_parse_url' )->alias( function( $link ) {
+                return parse_url( $link );
+            } );
```
Comment (remyperona) [inline tests/Unit/inc/Engine/Media/AboveTheFold/WarmUp/Controller/fetchLinks.php]:
```suggestion
            $this->stubWpParseUrl();
```

File: tests/Unit/inc/Engine/License/API/User/hasLicenseExpiredPast15Days.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace WP_Rocket\tests\Unit\inc\Engine\License\API\User;
```
Comment (remyperona) [inline tests/Unit/inc/Engine/License/API/User/hasLicenseExpiredPast15Days.php]:
```suggestion
namespace WP_Rocket\Tests\Unit\Inc\Engine\License\API\User;
```

File: tests/Unit/inc/Engine/License/API/User/hasLicenseExpiredPast15Days.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+use WP_Rocket\Engine\License\API\User;
+use WP_Rocket\Tests\Unit\TestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\License\API\User::has_license_expired_more_than_15_days
+ *
+ * @group License
+ */
+class hasLicenseExpiredPast15Days extends TestCase {
```
Comment (remyperona) [inline tests/Unit/inc/Engine/License/API/User/hasLicenseExpiredPast15Days.php]:
```suggestion
class Test_hasLicenseExpiredPast15Days extends TestCase {
```

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Comment by Mai-Saad** [comment]
related test case https://wpmediaqa.testrail.io/index.php?/cases/view/15319 is already on [test Run](https://wpmediaqa.testrail.io/index.php?/runs/view/834&group_by=cases:section_id&group_order=asc) used at this PR https://github.com/wp-media/wp-rocket/pull/6465 (so if it passed there, we can move this PR too to done)

**Comment by MathieuLamiot** [comment]
Let's move it to Done then. If there is an issue during test execution, we'd open a dedicated GH issue anyway

**Pull request merged**